core/DataAccess/ArchiveSelector.php

```diff
@@ -338,6 +359,30 @@ private static function moveChunkRowToRows(&$rows, $row, Chunk $chunk, $loadAllS
         }
     }
 
+    public static function getLatestArchiveStartTimestampForToday($idSite, $period, $date, Segment $segment)
+    {
+        $today = Date::factoryInTimezone('today', Site::getTimezoneFor($idSite));
+        $table = ArchiveTableCreator::getNumericTable($today);
+
+        $periodObj = Period\Factory::build($period, $date);
+
+        $nameCondition = self::getNameCondition(['VisitsSummary'], $segment);
+
+        $sql = "SELECT MAX(idarchive)
+                  FROM $table
+                 WHERE idsite = ? AND period = ? AND date1 = ? AND date2 = ? AND $nameCondition";
+        $bind = [$idSite, $periodObj->getId(), $periodObj->getDateStart()->getDatetime(), $periodObj->getDateEnd()->getDatetime()];
+
+        $idArchive = Db::fetchOne($sql, $bind);
+
+        $sql = "SELECT ts_archived
+                  FROM $table
+                 WHERE idarchive = ? AND name = ?";
+        $bind = [$idArchive, ArchiveWriter::ARCHIVE_START_RECORD_NAME];
```

Comment:
Initially I was wondering if we only should look at archives where the archiving finished  in case the previous archiving had an error and interrupted (same idArchive should also have the done flag with status OK or temporary). 

Then realised we can likely not do this because the current archiving might be in progress and that's the reason why the done flag is missing. => This should not be a problem though because the archiving would then be simply skipped as the logic should notice for that period/date there is already an archiver running right now.

So what if the last archive stopped in the middle because of an error, then we would maybe not start that archive again under circumstances? Say start flag is `00:15am`, then the archiving broke in the middle, ... it would maybe not start the archiving again? Maybe, to play it safe, we might need to always check to only look at `ts_archived` for archives that have the start and the done flag.

Not sure I explain it well.

Reply:
My understanding was that's what we do, at least that's what `$nameCondition` is for. Ie, it should be `(name IN ... AND value IN (DONE_OK, DONE_OK_TEMPORARY)`. So we're only looking at done archives.

Incidentally I think this is also why we shouldn't append the suffix to the `start` record, since we'd have to check the done flag value anyway.

Reply:
I know we've all said it before, but the archiving process confuses the heck out of me.

Reply:
> I know we've all said it before, but the archiving process confuses the heck out of me.

me too :)

Reply:
> My understanding was that's what we do, at least that's what $nameCondition is for. Ie, it should be (name IN ... AND value IN (DONE_OK, DONE_OK_TEMPORARY). So we're only looking at done archives.

Makes sense . Just need to double check that we also look at that in the `CronArchive`

Reply:
CronArchive uses `ArchiveSelector::getLatestArchiveStartTimestampForToday()` which uses the name condition, so I think we're good here.